erpnext/payroll/doctype/gratuity/gratuity.json

```diff
@@ -162,41 +156,48 @@
    "fieldtype": "Column Break"
   },
   {
-   "depends_on": "eval: doc.pay_via_salary_slip == 1",
+   "depends_on": "eval: doc.payment_method == \"Additional Salary\"",
    "fieldname": "payroll_date",
    "fieldtype": "Date",
    "label": "Payroll Date",
-   "mandatory_depends_on": "eval: doc.pay_via_salary_slip == 1"
+   "mandatory_depends_on": "eval: doc.payment_method == \"Additional Salary\""
   },
   {
    "default": "0",
-   "depends_on": "eval:doc.pay_via_salary_slip == 0",
+   "depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "fieldname": "paid_amount",
    "fieldtype": "Currency",
    "label": "Paid Amount",
    "read_only": 1
   },
   {
-   "depends_on": "eval: doc.pay_via_salary_slip == 0",
+   "depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "fieldname": "payable_account",
    "fieldtype": "Link",
    "label": "Payable Account",
-   "mandatory_depends_on": "eval: doc.pay_via_salary_slip == 0",
+   "mandatory_depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "options": "Account"
   },
   {
-   "depends_on": "eval: doc.pay_via_salary_slip == 0",
+   "depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "fieldname": "cost_center",
    "fieldtype": "Link",
    "label": "Cost Center",
-   "mandatory_depends_on": "eval: doc.pay_via_salary_slip == 0",
+   "mandatory_depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "options": "Cost Center"
+  },
+  {
+   "default": "Additional Salary",
+   "fieldname": "payment_method",
+   "fieldtype": "Select",
+   "label": "Payment Method",
+   "options": "Additional Salary\nPayment Entry"
```

Comment:
We will need a patch to update the payment method in existing documents?

Reply:
done